specs/0121-translation-memory-management.md

```diff
@@ -0,0 +1,43 @@
+- Feature Name: Translation Memory Management
+- Created: 2024-08-28
+- Associated Issue: #3031
+
+# Summary
+
+Add ability to edit, delete and upload Translation Memory (TM) entries.
+
+# Motivation
+
+Translation memory is a fundamental tool for faster and more consistent translations, and has become even more important with pretranslation, since it's used to train custom machine translation engines in Google AutoML Translation.
+
+Support for TM in Pontoon is limited:
+* Localizers cannot edit or delete existing TM entries directly.
+* TM entries can be changed indirectly via translation workspace, but only if the corresponding strings are still available for translation.
+* Localizers cannot upload their own translation memories.
+
+# Feature explanation
+
+A new "Translation Memory" tab is added to the Team pages, which lists all TM entries for the team. For each entry, source string and translation are provided, clicking on which leads to the translation workspace for the corresponding strings (using the list `query` parameter). Entries are listed in alphabetic order and only shown once for each unique combination of source string and translation.
+
+Initially, only the first 100 entries are loaded. Additional entries are loaded in batches of 100 using infinite scroll, a UX pattern that is also used in the string list in translation workspace.
+
+TM entries can be searched by the source string and translation using the Search field. Search is triggered by pressing Enter or 500 ms after the user has stopped typing.
```

Comment:
Why the 500ms? I think it doesn't match the UX in the rest of Pontoon, and it makes sense to react only to enter.

Reply:
True, it doesn't match the rest of Pontoon, but requiring Enter also doesn't. The search field in the Projects tab looks exactly the same and searches as you type. 

We can't have that in TM, because it needs to be server side due to the much bigger amount of data.

Should I just drop the 500 ms part and file a bug acknowledging the inconsistency?

Reply:
While they're closer, I think the comparison shouldn't be with the project tab, but with the search in Translate view. I'm OK with dropping and aknowledging.

Reply:
OK, I'll drop the 500 ms part.

(The reason why I'm bringing up the Project tab is because the widgets have the exact same designs. For a good UX, it's important that widgets that look the same also behave the same to avoid confusion and maintain usability.)